stout/http.h

```diff
@@ -0,0 +1,524 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+#include "curl/curl.h"
+#include "stout/event-loop.h"
+#include "stout/eventual.h"
+#include "uv.h"
+
+////////////////////////////////////////////////////////////////////////
+
+namespace stout {
+namespace eventuals {
+namespace http {
+
+////////////////////////////////////////////////////////////////////////
+
+// Used for application/x-www-form-urlencoded case.
+// TODO(folming): switch to RapidJSON.
+using PostFields = std::unordered_map<std::string, std::string>;
+
+////////////////////////////////////////////////////////////////////////
+
+struct Result {
+  std::string data;
+  long response;
+};
+
+////////////////////////////////////////////////////////////////////////
+
+namespace detail {
```

Comment:
I believe we've been moving `stout-eventuals` towards more class-based separation instead of using a `detail` namespace. How would you feel about having an `HTTP` class with a private `Call` method, e.g. `HTTP::Get()` (public) calling `HTTP::Call()` (private)?